src/main/java/org/openrewrite/java/spring/security5/UseNewRequestMatchers.java

```diff
@@ -0,0 +1,110 @@
+
+    private static final MethodMatcher ANT_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry antMatchers(..)");
+    private static final MethodMatcher MVC_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry mvcMatchers(..)", true);
+    private static final MethodMatcher REGEX_MATCHERS = new MethodMatcher("org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry regexMatchers(..)");
+
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Use the new `requestMatchers` methods";
+    }
+
+    @NonNull
+    @Override
+    public String getDescription() {
+        return "In Spring Security 5.8, the `antMatchers`, `mvcMatchers`, and `regexMatchers` methods were deprecated in favor of new `requestMatchers` methods. Refer to the [Spring Security docs](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#use-new-requestmatchers) for more information.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return Applicability.or(
+                new UsesMethod<>(ANT_MATCHERS),
+                new UsesMethod<>(MVC_MATCHERS),
+                new UsesMethod<>(REGEX_MATCHERS));
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                J.MethodInvocation mi = super.visitMethodInvocation(method, ctx);
+                if (ANT_MATCHERS.matches(mi) || MVC_MATCHERS.matches(mi) || REGEX_MATCHERS.matches(mi)) {
+                    mi = maybeChangeMethodInvocation(mi);
+                }
+                return mi;
+            }
+        };
+
+    }
+
+    private J.MethodInvocation maybeChangeMethodInvocation(J.MethodInvocation mi) {
+        JavaType.Method requestMatchersMethod = findRequestMatchersMethodWithMatchingParameterTypes(mi);
+        if (requestMatchersMethod != null) {
+            return mi
+                    .withMethodType(requestMatchersMethod)
+                    .withName(mi.getName().withSimpleName("requestMatchers"));
+        } else {
+            return mi;
+        }
+    }
+
+    @Nullable
+    private JavaType.Method findRequestMatchersMethodWithMatchingParameterTypes(J.MethodInvocation mi) {
+        JavaType.Method methodType = requireNonNull(mi.getMethodType(), "methodType");
+        List<JavaType> parameterTypes = methodType.getParameterTypes();
+        List<JavaType.Method> methods;
+        boolean isOverride = TypeUtils.isOverride(mi.getMethodType());
+        if (isOverride) {
+            methods = requireNonNull(methodType.getDeclaringType().getSupertype(), "superType").getMethods();
```

Comment:
Not sure you'd need to look for overrides... The methods you're matching and the replacement method all declared in the same type... Therefore, line 102 alone should handle everything correctly... Don't think you'd need to go up to the super type from the declaring type

Reply:
I thought so too but without this I could not find a `requestMatchers` method in the `mvcMatchers` cases.

In my head it makes sense because: `mvcMatchers` [is abstract](https://docs.spring.io/spring-security/site/docs/5.8.x/api/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.html#mvcMatchers(java.lang.String...)) so any methodInvocation will always be on a subclass of `AbstractRequestMatcherRegistry`.

Without looking at the supertype I could not find the `requestMatchers` method.